homeassistant/components/bluesound/media_player.py

```diff
@@ -190,12 +227,10 @@ async def async_service_handler(service: ServiceCall) -> None:
         }
         if entity_ids := service.data.get(ATTR_ENTITY_ID):
             target_players = [
-                player
-                for player in hass.data[DATA_BLUESOUND]
-                if player.entity_id in entity_ids
+                player for player in hass.data[DOMAIN] if player.entity_id in entity_ids
```

Comment:
Side note: In the future this should be refactored to not store the entities in hass.data but use our entity service helper to register entity services, and also use our standard ways to group media player entities (which doesn't require custom services).

https://developers.home-assistant.io/docs/core/entity/media-player#grouping-player-entities-together

To keep track of added entities we can eg store the player id. Look at the heos integration for an example how to map players to media player entities and group media players.

Reply:
I already have a branch for implementing the standard way to group media player entities. My solution was:

- get `entity_entry` from `entity_registry`
- get `config_entry` from `hass.config_entries.async_get_entry(entity_entry.config_entry_id)`
- the config_entry has everything needed for grouping

But there is one big problem: I have only one Player. I cannot test it.

Reply:
Are there any examples for `entity services`? I couldn't find any.

Reply:
Here's an example:
https://github.com/home-assistant/core/blob/8bc9fd23bbe9d7ddcc3d190b327cdb4a2dc4521b/homeassistant/components/denonavr/media_player.py#L142-L147

Reply:
> But there is one big problem: I have only one Player. I cannot test it.

Written tests are enough, if you can't test it manually.

Reply:
I have a few players and I'm happy to help with testing over the next week or two. Can you point me to the release or branch that needs to be tested, and give a brief description of the testing that you'd like?

Reply:
That would be great. I don't have working code yet. And I have 2 things i want to do first before refactoring the media player grouping. I can tag you on the media player grouping PR once I have created it, but it will most likely be more that two weeks from now.